cligen_cv.c

```diff
@@ -2003,24 +2003,24 @@ str2time(const char           *in,
  *   char timestr[28];
  *   struct timeval tv;
  *   gettimeofday(&tv);
- *   if (time2str(tv, timestr, sizeof(timestr)) < 0)
+ *   if (time2str(&tv, timestr, sizeof(timestr)) < 0)
  *     err;
  *   printf("%s", timestr);
  * @endcode
  * @see str2time
  */
 int
-time2str(struct timeval tv, 
+time2str(const struct timeval *tv,
          char          *fmt,
-         int            len)
+         unsigned       len)
 {
     int        retval = -1;
     struct tm *tm;
 
-    tm = gmtime((time_t*)&tv.tv_sec);
+    tm = gmtime((time_t*)tv->tv_sec);
```

Comment:
Actually this should be: tm = gmtime((time_t*)&tv->tv_sec);

Reply:
I dont know. There are many systems out there, it might have been added for a reason.

Reply:
`a5560d93 cligen_var.c (Olof Hagsand 2013-07-12 15:42:06 +0200 2020)     tm = gmtime((time_t*)&tv.tv_sec);`
It's been there a decade+

Reply:
I remember tv_sec could be signed in some archs

Reply:
Yikes. Negative time.